### PR TITLE
9002 Allow to choose color ramps for discrete features

### PR DESCRIPTION
**Pull request by IagoLast**
https://github.com/CartoDB/cartodb/issues/9002
---

- [x] Add tests
- [x] Use colors from `cartocolor`
- [x] Remove dead code & unused events ?
- [x] Refactor code
- [x] Use StackView properly

### CODE REVIEW

**Comment by matallo** [comment]
NEWS.md and choose a more descriptive title! 🙏

File: NEWS.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ This release changes the way Google ouath login works. If you are using it, you
 to the oauth.google_plus section of the configuration file.
 
 ### Features
+* Add customizable color ramps for qualitative attributes ([#9002](https://github.com/CartoDB/cartodb/issues/9002))
```
Comment (matallo) [inline NEWS.md]: 🤗

Comment (jesusbotella) [reply NEWS.md]: ❤️

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/categories-list/list-item-view/categories-list-item-view.js [hunk 2]
```diff
@@ -0,0 +1,60 @@
+var _ = require('underscore');
```
Comment (ivanmalagon) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/categories-list/list-item-view/categories-list-item-view.js]: Why haven't you used `cartodb3/components/form-components/editors/fill/input-color/input-categories/categories-list/categories-list-item-view.js`.

It's the same code so I guess you did some copypasta here. Why is it duplicated?

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/categories-list/list-item-view/categories-list-item.tpl [hunk 3]
```diff
@@ -0,0 +1,19 @@
+<div class="CDB-ListDecoration-itemLink <% if (isSelected) { %> is-selected <% } %>">
```
Comment (ivanmalagon) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/categories-list/list-item-view/categories-list-item.tpl]: Same copypasta here with `cartodb3/components/form-components/editors/fill/input-color/input-categories/input-color-categories-list-item.tpl`

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/categories-list/list-view/categories-custom-list-view.js [hunk 4]
```diff
@@ -0,0 +1,22 @@
+var CustomListView = require('cartodb3/components/custom-list/custom-list-view');
```
Comment (ivanmalagon) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/categories-list/list-view/categories-custom-list-view.js]: Same copypasta here with `cartodb3/components/form-components/editors/fill/input-color/input-categories/categories-list/categories-list-view.js`

Too much duplication I see here.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-value-content-view.js [hunk 5]
```diff
@@ -55,28 +57,20 @@ module.exports = CoreView.extend({
   },
```
[the remark this answers is not in the document]
Comment (ivanmalagon) [reply lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-color-value-content-view.js]: Forget about this. I've already push `module:` info to the views of this PR.

Just remember it next time you touch a view file. You can add this info to ease finding the view in the DOM.

File: lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/main-view.js [hunk 6]
```diff
@@ -1,17 +1,23 @@
 var _ = require('underscore');
-var CDB = require('cartodb.js');
+var cdb = require('cartodb.js');
 var Backbone = require('backbone');
 var CoreView = require('backbone/core-view');
-var template = require('./input-color-categories.tpl');
+var MainViewTemplate = require('./main-view.tpl');
```
Comment (matallo) [inline lib/assets/javascripts/cartodb3/components/form-components/editors/fill/input-color/input-qualitative-ramps/main-view.js]: templates should be camelCase, starting with lowercase

(I'll take this small detail)

File: lib/assets/locale/index.js [hunk 7]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
```
Comment (matallo) [inline lib/assets/locale/index.js]: what's up with these locales? I'm removing them

**Comment by jesusbotella** [comment]
CAN'T BELIEVE IT! VAMOS! 💪 🎉

**Pull request merged**